lib/web/apiserver.go

```diff
@@ -1930,11 +1930,11 @@ func (h *Handler) installer(w http.ResponseWriter, r *http.Request, p httprouter
 
 	tmpl := installers.Template{
```

Comment:
I considered moving the quoting to these delegate functions, however I was concerned about it causing a more significant change.

In the current places we are doing the quoting directly where we are pulling the values from a user supplied location.  If we move the logic to `installer` and `getJoinScript`, we need to assume that potentially any parameter may have been user supplied.  That is not always true, so we end up quoting some parameters which were programmatically supplied.

So in general I agree I want to put this logic where it's less likely to be forgotten.  But it does belong close to where we are taking in the inputs from a user, since application filled inputs are safe.

That said, I think we can change it either way, but this is why I have it implemented where we quote directly when we are accepting user inputs.  Let me know your thoughts.

Reply:
To more concretely demonstrate it, if we wanted to move this logic `getJoinScript`, I believe we would need to quote most of these fields:
https://github.com/gravitational/teleport/blob/master/lib/web/join_tokens.go#L499-L522

That would address the concern you have, but I am concerned about it being a larger change.  I think it makes sense to keep the quoting close to where we are accepting user supplied values.  And as part of general process we should always be vigilant in validating user supplied values.

I don't feel strongly, I just wanted to make sure it's understood why I was trying to minimize this change.  If you feel strongly I will update it to quote the fields there.

Reply:
Probably not :shrug:

Reply:
+1 on quoting as close to generating the shell script as possible
We should be able to blindly apply the quoting to all the map's values.

Reply:
Thank you for the feedback @rosstimothy and @marcoandredinis


I updated the PR so that we are quoting in `getJoinScript` instead of when we are pulling the parameters.  We can't do the same with `instTmpl.Execute` as this is part of go's toolchain.